[Flatpak] Avoid error: Invalid byte sequence in conversion input

calvaris opened the pull request:
#### c95d06d44224ecba65a0c9d5609f49ac97eabc53
<pre>
[Flatpak] Avoid error: Invalid byte sequence in conversion input
<a href="https://bugs.webkit.org/show_bug.cgi?id=249464">https://bugs.webkit.org/show_bug.cgi?id=249464</a>

Reviewed by Philippe Normand.

Check for the existence of the en_US.utf8 locale to enable the fancy prompt.

* Tools/flatpak/flatpakutils.py:
(WebkitFlatpak.run_in_sandbox):

Canonical link: <a href="https://commits.webkit.org/258774@main">https://commits.webkit.org/258774@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fe078914fb9659d61a79bf77ea9ae7882175b15c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/102747 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/11868 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/35775 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/112004 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/172234 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/106713 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/12880 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/2777 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/95000 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/109685 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/108522 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/9880 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/93092 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/37527 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/91732 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/24610 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/79267 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/5323 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/26027 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/5473 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/2476 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/11493 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/45519 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6033 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/7215 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

philn commented on Tools/flatpak/flatpakutils.py:
```diff
@@ -817,7 +817,12 @@ def run_in_sandbox(self, *args, **kwargs):
+                    if 'en_US.utf8' in subprocess.check_output(['locale', '-a']).decode("utf8").split():
+                        sandbox_environment["PS1"] = f"[📦🌐🐱 $FLATPAK_ID {self.platform}@{self.build_type} \\W]\\$ "
+                except:
+                    void
```
`void`? I don't think this is a valid python thing. Should be `pass`.

philn commented on Tools/flatpak/flatpakutils.py:
```diff
@@ -817,7 +817,12 @@ def run_in_sandbox(self, *args, **kwargs):
-                sandbox_environment["PS1"] = f"[📦🌐🐱 $FLATPAK_ID {self.platform}@{self.build_type} \\W]\\$ "
+                sandbox_environment["PS1"] = f"[$FLATPAK_ID {self.platform}@{self.build_type} \\W]\\$ "
+                try:
+                    if 'en_US.utf8' in subprocess.check_output(['locale', '-a']).decode("utf8").split():
```
Any utf8 locale could be used, so no need to gate this on the en_US one.

calvaris replied:
No, I have gl_ES.utf8 as default and it does not work if I don't have en_US.utf8 available

TingPing commented on Tools/flatpak/flatpakutils.py:
```diff
@@ -817,7 +817,12 @@ def run_in_sandbox(self, *args, **kwargs):
+                try:
+                    if 'en_US.utf8' in subprocess.check_output(['locale', '-a']).decode("utf8").split():
+                        sandbox_environment["PS1"] = f"[📦🌐🐱 $FLATPAK_ID {self.platform}@{self.build_type} \\W]\\$ "
+                except:
```
A blank except is bad form in general, it catches everything like hitting ctrl+c.

You probably want `except subprocess.CalledProcessError:`

You can test it by just running incorrect commands, `['fakecommand']`, or `['locale', '--totallynotaflag']` to get the real world exceptions.

calvaris replied:
It was `FileNotFoundError`.

calvaris committed:
[Flatpak] Avoid error: Invalid byte sequence in conversion input
https://bugs.webkit.org/show_bug.cgi?id=249464

Reviewed by Philippe Normand.

Check for the existence of the en_US.utf8 locale to enable the fancy prompt.

* Tools/flatpak/flatpakutils.py:
(WebkitFlatpak.run_in_sandbox):

Canonical link: https://commits.webkit.org/258774@main
